src/XML/src/FMI3/fmi3_xml_parser.c

```diff
@@ -251,17 +251,17 @@ void fmi3_xml_parse_free_context(fmi3_xml_parser_context_t *context) {
         context->parser = 0;
     }
     fmi3_xml_free_parse_buffer(context);
-    if(context->attrMap) {
-        jm_vector_free(jm_named_ptr)(context->attrMap);
-        context->attrMap = 0;
+    if(context->attrMapByName) {
+        jm_vector_free(jm_named_ptr)(context->attrMapByName);
+        context->attrMapByName = 0;
     }
     if(context->elmMap) {
         jm_vector_free(fmi3_xml_element_handle_map_t)(context->elmMap);
         context->elmMap = 0;
     }
-    if(context->attrBuffer) {
-        jm_vector_free(jm_string)(context->attrBuffer);
-        context->attrBuffer = 0;
+    if(context->attrMapById) {
```

Comment:
```suggestion
    if (context->attrMapById) {
```
also on L258 but unable to do suggestion because no change was made here.